lib/rules/unicode-bom.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileoverview Allow or disallow Unicode BOM
+ * @author Andrew Johnston <https://github.com/ehjay>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "allow or disallow Unicode BOM",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+
+        schema: [
+            {
+                enum: ["always", "never"]
+            }
+        ]
+    },
+
+    create: function(context) {
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        return {
+
+            Program: function checkUnicodeBOM(node) {
+
+                var sourceCode = context.getSourceCode(),
+                    location = {column: 1, line: 1},
+                    requireBOM = context.options[0] || "never";
+
+                if (!sourceCode.hasBOM && (requireBOM === "always")) {
+                    context.report({
+                        node: node,
+                        loc: location,
+                        message: "Missing unicode BOM (Byte Order Mark)."
+                    });
```

Comment:
I love auto fixing!
Could you add `fix` function?

For `"always"`:

```js
fix: function(fixer) {
    return fixer.insertTextBefore(node, "\uFEFF");
}
```

For `"never"`:

```js
fix: function(fixer) {
    return fixer.removeRange([-1, 0]);
}
```